x-pack/plugins/ingest_pipelines/public/application/pipeline_editor/components/processor_settings_form/processors/gsub.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import {
+  FieldConfig,
+  FIELD_TYPES,
+  fieldValidators,
+  FormRow,
+  UseField,
+  Field,
+} from '../../../../../shared_imports';
+
+const { emptyField } = fieldValidators;
+
+const fieldConfig: FieldConfig = {
+  type: FIELD_TYPES.TEXT,
+  label: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.fieldFieldLabel', {
+    defaultMessage: 'Field',
+  }),
+  validations: [
+    {
+      validator: emptyField(
+        i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.fieldRequiredError', {
+          defaultMessage: 'A field value is required.',
+        })
+      ),
+    },
+  ],
+};
+
+const patternConfig: FieldConfig = {
+  type: FIELD_TYPES.TEXT,
+  label: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.patternFieldLabel', {
+    defaultMessage: 'Pattern',
+  }),
+  validations: [
```

Comment:
Not sure if this field is for index pattern or not, but in case it is, know that there is an `indexPatternField` validator for them.

https://github.com/elastic/kibana/blob/master/src/plugins/es_ui_shared/static/forms/helpers/field_validators/index_pattern_field.ts#L26